test/tasks/fixtures/index.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  taskMiddleware,
+  withTask,
+  taskCreator,
+  drainTasksForTesting,
+  Task,
+} from '../../../src/tasks';
+
+import {createStore, applyMiddleware} from 'redux';
+
+export const ECHO_TASK = taskCreator<string, string>((payload, success) =>
+  success(payload), 'ECHO_TASK');
+
+// sync set task
+export const SET_TASK_SYNC = taskCreator((payload, success) => success(payload), 'SET_TASK_SYNC');
+
+// async set task
+export const SET_TASK_ASYNC = taskCreator((payload, success) =>
+  new Promise(resolve => {
+    setTimeout(() => {
+      success(payload);
+      resolve();
+    }, 0);
```

Comment:
yes, I like the second arg. Ideally we use something like `setImmediate` but I don't think that's a thing yet.